packages/aws-cdk-lib/aws-s3/test/rules.test.ts

```diff
@@ -334,4 +334,163 @@ describe('rules', () => {
       },
     });
   });
+
+  describe('required properties for rules', () => {
+    test('throw if there is a rule doesn\'t have required properties', () => {
+      const app = new App();
+      const stack = new Stack(app);
+      new Bucket(stack, 'MyBucket', {
+        lifecycleRules: [
+          {
+            objectSizeLessThan: 300000,
+            objectSizeGreaterThan: 200000,
+          },
+        ],
+      });
+      expect(() => {
+        app.synth();
```

Comment:
Because rendering is done as Lazy:

https://github.com/aws/aws-cdk/blob/v2.162.1/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L1941